include/kodlab_mjbots_sdk/mjbots_hardware_interface.h

```diff
@@ -103,7 +104,14 @@ class MjbotsHardwareInterface  {
    * @brief accessor for the IMU data of the robot
    * @return const IMU data shared pointer for the robot
    */
-  const std::shared_ptr<::kodlab::IMUData<float>> GetIMUDataSharedPtr();
+  const std::shared_ptr<::kodlab::IMUData<float>> GetIMUDataSharedPtr();        
+  
+  /*!
+  * @brief Setter for the robot's IMU data pointer. Releases the previously owned IMU data object
+  *
+  * @param imu_data_ptr a shared pointer to kodlab::IMUData
+  */
+  void SetIMUDataSharedPtr(std::shared_ptr<::kodlab::IMUData<float>> imu_data_ptr){imu_data_ = imu_data_ptr;}
```

Comment:
Added a setter so that the user can define this however they want, included incorrectly